tasks/task6/schedule/schedule.py

```diff
@@ -0,0 +1,67 @@
+import time
+from sender import Sender
+from plataform import MessagePlataform
+
+class Schedule(Sender):
+    """
+    Class Schedule that allows to send/receive messages in a specific time
+    """
+    def __init__(self,hours=0,minutes=0,seconds=0):
+        super().__init__()
+        self.hours = hours
+        self.minutes = minutes
+        self.seconds = seconds
+        self.period = 0
+        self.plataforms = []
+        self.message = 'test message'
+        self.message_received = False
+        #  self.time_to_wait_in_seconds = 10 * 60
+        self.time_to_wait_in_seconds = 1
+
+    def set_time(self, hour, minute, second):
+        self.hours = hour
+        self.minutes = minute
+        self.seconds = second
+        self.period = 0
+        
+    def set_time_by_period(self,period):
+        self.period = period
+        self.start_time = False
+    
+    def add_plataforms(self,plataform : MessagePlataform):
+        plataform.set_sender(self)
+        self.plataforms.append(plataform)
+    
+    def set_message(self,message):
+        self.message = message
+
+    def run(self):
+        while self.period:
+            for plataform in self.plataforms:
+                plataform.receive_message(self.message)
+                time.sleep(self.time_to_wait_in_seconds)
+                if self.message_received:
+                    return
+        self._sleep()           
+        for plataform in self.plataforms:
```

Comment:
Wouldn't this part be duplicated, and it seems that it would be doing the same thing that is already done in lines 40 to 44 ? or is this handling another scenario?